src/connection/utils.ts

```diff
@@ -12,18 +12,18 @@ export function getIsInjected(): boolean {
   return Boolean(window.ethereum)
 }
 
+export function getIsBraveWallet(): boolean {
+  return window.ethereum?.isBraveWallet ?? false
+}
+
 export function getHasMetaMaskExtensionInstalled(): boolean {
-  return window.ethereum?.isMetaMask ?? false
+  return (window.ethereum?.isMetaMask ?? false) && !getIsBraveWallet()
```

Comment:
This logic breaks down when a Brave wallet has Metamask installed. In that case, it just picks up the MetaMask wallet 🤔 

We need well-defined requirements from product as to how to handle that case. Most likely, we need to support extra wallet options in our [wallet options](https://github.com/Uniswap/interface/blob/main/src/components/WalletModal/index.tsx) component. It seems to be in this ticket's scope to also handle this case.

Reply:
When Brave wallet has MetaMask installed and enabled (as a preferred extension), `isBraveWallet` will be false. `isBraveWallet` is only true when users explicitly decided to use Brave as their primary extension.

Reply:
Here's a breakdown from my manual testing of this:
- **MetaMask installed, Brave Wallet disabled ([instructions](https://metamask.zendesk.com/hc/en-us/articles/360038596792-Using-MetaMask-wallet-in-Brave-browser))** - connects with MetaMask and displays MetaMask 
- **MetaMask not installed, Brave wallet disabled** - Install MetaMask
- **MetaMask not installed, Brave wallet enabled** - Connect with Browser Wallet
- **MetaMask installed, Brave Wallet enabled** - Depending on user settings - either MetaMask (if MetaMask is selected as “default”, even though Brave Wallet is enabled) or Browser Wallet (in case MetaMask is installed, but not selected as default).

Reply:
You're right, all those cases are correctly covered! It just seemed super confusing for me from a user's standpoint. Feels like we should be able to support both, but it does not seem so from Brave's documentation 🤷